docs/sources/reference/builder.md

```diff
@@ -350,8 +350,9 @@ accessible from the host by default. To expose ports to the host, at runtime,
 
 The `ENV` instruction sets the environment variable `<key>` to the value
 `<value>`. This value will be passed to all future 
-`RUN`, `ENTRYPOINT`, and `CMD` instructions. This is
-functionally equivalent to prefixing the command with `<key>=<value>`
+`RUN`, `ENTRYPOINT`, and `CMD` instructions. To set a value for a single
+command without affecting future commands or containers, prefix the command
+with `<key>=<value>`.
```

Comment:
Yeah that’s my fault. :)

If you want to link/reorganize the two sections, I won’t object, but Sven or Fred might. :P

> On Jan 22, 2015, at 6:47 PM, Michael A. Smith <notifications@github.com> wrote:
> 
> In docs/sources/reference/builder.md <https://github.com/docker/docker/pull/10273#discussion_r23428928>:
> 
> > @@ -350,8 +350,9 @@ accessible from the host by default. To expose ports to the host, at runtime,
> >  
> >  The `ENV` instruction sets the environment variable `<key>` to the value
> >  `<value>`. This value will be passed to all future 
> > -`RUN`, `ENTRYPOINT`, and `CMD` instructions. This is
> > -functionally equivalent to prefixing the command with `<key>=<value>`
> > +`RUN`, `ENTRYPOINT`, and `CMD` instructions. To set a value for a single
> > +command without affecting future commands or containers, prefix the command
> > +with `<key>=<value>`.
> @duglin <https://github.com/duglin> It may not be obvious to everyone that you can set a value in the environment just for a single command. And once you set an ENV you can't really unset it. So I still think RUN foo=bar do-something should exist. But maybe it doesn't belong here. Maybe it belongs in the best practices <https://github.com/docker/docker/blob/master/docs/sources/articles/dockerfile_best-practices.md>.
> 
> Also, I realized the Environment Replacement section starting on line 108 covers your original question. Maybe it just needs to be better tied to the actual ENV documentation.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/docker/docker/pull/10273/files#r23428928>.
>